kafka-sync.go

```diff
@@ -299,31 +302,54 @@ func (s *Syncer) IndexTopic(kafka sarama.Client, index diff.Index) (msgCount uin
 		}
 	}
 
-	msgCount = 0
-	for m := range pc.Messages() {
-		hw := pc.HighWaterMarkOffset()
-		if hw > highWater {
-			highWater = hw
-		}
-		glog.V(4).Info("-> offset: ", m.Offset, " / ", highWater-1)
-
-		value := m.Value
-		if bytes.Equal(value, s.RemovedValue) {
-			value = nil
+	// report kafka errors
+	go func() {
+		for m := range pc.Errors() {
```

Comment:
Add comment // FIXME fail on first error

Reply:
did that and will try to work on this specific FIXME after working on an amazing sharepoint2sync middleware app.